src/test/java/io/openliberty/tools/intellij/it/TestUtils.java

```diff
@@ -540,5 +540,25 @@ public static void checkDebugPort(String absoluteWLPPath, int debugPort) throws
         Assertions.assertTrue(debugPortIsSet, "Debug Port is not set to " + debugPort);
     }
 
+    /**
+     * Copies all files and directories from the source directory to the destination directory.
+     *
+     * @param sourceDirectoryLocation The path to the source directory to copy from.
+     * @param destinationDirectoryLocation The path to the destination directory to copy to.
+     * @throws IOException If an I/O error occurs while copying files.
+     */
+    public static void copyDirectory(String sourceDirectoryLocation, String destinationDirectoryLocation)
+            throws IOException {
+        Files.walk(Paths.get(sourceDirectoryLocation))
+                .forEach(source -> {
+                    Path destination = Paths.get(destinationDirectoryLocation, source.toString()
+                            .substring(sourceDirectoryLocation.length()));
+                    try {
```

Comment:
Why do we need to catch this exception but not the one from `Files.walk()`? The test can't proceed if any files are missing and if it does a strange failure will result. You should remove this `try/catch`.

Reply:
@turkeylurkey , I found this code from [here ](https://www.baeldung.com/java-copy-directory#using-the-javanio-api) 
I tried removing the `try-catch` block from here and adding it from `Files.walk()`, but I'm getting an error on this line. The quick-fix suggests adding a `try-catch` block for the following line: `Files.copy(source, destination);`

<img width="696" alt="Screenshot 2024-08-16 at 5 35 48 PM" src="https://github.com/user-attachments/assets/3a7597e7-1409-4cc6-be9a-cddc343116ec">


Since we throw an `IOException` for the `copyDirectory(String sourceDirectoryLocation, String destinationDirectoryLocation)` method, I believe that the exception will definitely be caught if any I/O operation fails and I believe the actual operation is performing at this line `Files.copy(source, destination);`

Reply:
I believe an IOException on line 557 will be caught and this method will complete normally with just a stack trace as evidence there was an error. I think it would be better if the test failed. Could you throw a RuntimeException or some other type to indicate this error when you catch IOException?

Reply:
@turkeylurkey, I made the change , please have a look

Reply:
This looks good but have you tested it? Could you mock a failed file copy and show us what the results and the logs look like?

Reply:
@turkeylurkey , I am able to reproduce the exception by adding the same line `Files.copy(source, destination)`  again as shown in the image below

<img width="1379" alt="Screenshot 2024-08-29 at 11 37 27 AM" src="https://github.com/user-attachments/assets/e94289e1-119f-4bba-9526-282b332875a5">

Reply:
👍